### DISCUSSION TITLE
Order of dimensions for ensemble model output

### DISCUSSION THREAD

**Discussion by knutfrode** [cf-convention/discuss#423] [Q&A about using CF] [answered]
As far as I can understand, CF-convention has no restrictions on the order of dimensions, but recommends using the order
`(time, depth/height/z, y, x)`

But is there a recommendation/requirement for the order of the additional ensemble/realization dimension for output from ensemble ocean or atmospheric models?
E.g. `(ensemble_number, time, depth/height/z, y, x)` or `(time, depth/height/z, ensemble_number, y, x)` ?

**Comment by davidhassell** [comment]
Hello @knutfrode,

I don't know of any such recommendation or requirement, but I personally would go for `(ensemble_number, time, depth/height/z, y, x)` on the grounds that, depending on your access patterns, netCDF may be faster when the dimensions are ordered from slowest moving to fastest moving, left to right.

**Comment by sethmcg** [comment, marked as answer]
There is such a recommendation in the second paragraph of section 2.4.  "All other dimensions should, whenever possible, be placed to the left of the spatiotemporal dimensions."  This is in accordance with the COARDS recommendation for ordering of additional dimensions.

**Comment by ChrisBarker-NOAA** [comment]
>  "All other dimensions should, whenever possible, be placed to the left of the spatiotemporal dimensions."

I think that settles it -- but might it be worth adding text specifically about ensembles?

NOTE, as for: 

> depending on your access patterns, netCDF may be faster when the dimensions are ordered from slowest moving to fastest moving, left to right.

With modern netcdf (e.g. HDF, zarr, ....) this would be controlled by how it's chunked -- anyone know of the top of their head what the netcdf default chunking would do in this case?

Of course, with everything chunky -- what you should do depends how you want to access the data -- if you want results fro all of the ensemble members at one place over time, you wouldn't want it chunked in the default way ...

**Reply by ChrisBarker-NOAA** [reply]
now that I think a bit more about it -- given chunking -- folks should always use the recommended ording of dimensions, and they can chunk how they need to match their access patterns.

**Reply by sethmcg** [reply]
We could mention ensembles as an example: "All other dimensions (e.g., an ensemble dimension) should, whenever possible, be placed to the left of the spatiotemporal dimensions."
